### ISSUE TITLE
receiving asset a second time from a different consignment "erases" old one

### ISSUE THREAD

**Issue by zoedberg** [RGB-WG/rgb-wallet#101]
While using rgb-lib we found a bug that we've been able to reproduce with the rgb-sandbox in this [branch](https://github.com/nicbus/rgb-sandbox/tree/send_after_double_receive).
In the test we have 3 wallets, the 1st sends some assets (100) to the 2nd using a blinded UTXO, then it sends other assets (200) to the 2nd again but using a witness TX, finally the 2nd wallet tries to send some assets (100) to a 3rd wallet using a witness TX, explicitly picking the outpoint from the first send as the input outpoint.

The error we receive (trying to perform the 3rd send) is `DataError::OutpointUnknown` from `opouts_by_outpoints` (https://github.com/RGB-WG/rgb-wallet/blob/master/std/src/persistence/stock.rs#L644).
So the `contract_index` seems to forget that some assets have been received to a certain outpoint.
IMO the reason for this is that the `consume_consignment` method (the one called to accept the consignment) empties out the `contract_index` (https://github.com/RGB-WG/rgb-wallet/blob/master/std/src/persistence/stock.rs#L176).

This happens also if the 2nd send is performed with a blinded UTXO. While in order to see the bug the 3rd send always needs to be a witness one.
I think the reason for this is that in the `consign` method we call `opouts_by_outpoints` (the method throwing the error) only if one of the `seals` (the list of beneficiaries) is a `BuilderSeal::Revealed` one.

So summing up, it seems that in some cases receiving the same asset from consignments with different histories causes the node to forget about old assignments and remember only the ones from last accepted consignment.
